src/ethereum/berlin/vm/instructions/memory.py

```diff
@@ -85,26 +72,19 @@ def mstore8(evm: Evm) -> None:
         If `evm.gas_left` is less than
         `3` + gas needed to extend memory.
     """
-    # convert to Uint as start_position + size_to_extend can overflow.
-    start_position = Uint(pop(evm.stack))
+    # STACK
+    start_position = pop(evm.stack)
     value = pop(evm.stack)
-    # make sure that value doesn't exceed 1 byte
-    normalized_bytes_value = (value & U8_MAX_VALUE).to_bytes(1, "big")
 
-    memory_extend_gas_cost = calculate_gas_extend_memory(
-        evm.memory, start_position, U256(1)
-    )
-    total_gas_cost = u256_safe_add(
-        GAS_VERY_LOW,
-        memory_extend_gas_cost,
-        exception_type=OutOfGasError,
-    )
-    evm.gas_left = subtract_gas(evm.gas_left, total_gas_cost)
+    # GAS
+    extend_memory(evm, start_position, U256(1))
+    charge_gas(evm, GAS_VERY_LOW)
 
-    # extend memory and subtract gas for allocating 32 bytes of memory
-    extend_memory(evm.memory, start_position, U256(1))
+    # OPERATION
+    normalized_bytes_value = (value & U8_MAX_VALUE).to_bytes(1, "big")
```

Comment:
Using the `to_bytes` function seems off in some way. Should we create a `Uint8` type instead?

Reply:
We should probably send the write through the normal `memory_write()` call for consistency so we need to cast to a 1 byte long `Bytes`.

I guess we could use `Bytes([value & U8_MAX_VALUE])`.